architecture/src/androidLibMain/kotlin/navigation/Navigator.kt

```diff
@@ -264,12 +266,26 @@ class ActivityNavigator<A : NavigationAction<*>>(private val navigationMapper: (
 
     private fun navigateToBrowser(browserSpec: NavigationSpec.Browser) {
         assert(manager?.activity != null)
-        val activity = manager?.activity ?: return
-        val uri = Uri.parse(browserSpec.url.toURI().toString())
-        val intent = Intent(Intent.ACTION_VIEW, uri)
 
-        intent.resolveActivity(activity.packageManager)?.let {
-            activity.startActivity(intent)
+        when (browserSpec.viewType) {
+            NavigationSpec.Browser.Type.CustomTab -> {
+                val builder = CustomTabsIntent.Builder()
+                val customTabsIntent = builder.build()
+                customTabsIntent.intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
+                customTabsIntent.launchUrl(
+                    ApplicationHolder.applicationContext,
```

Comment:
This class can only meant to be used as a default argument, as it's usage of integrating it with your app is optional.

It seems context access can already be used by `manager?.activity` as elsewhere in the class.

Reply:
So I shouldn't implement the changes here? I didn't understand what did you mean.

> This class can only meant to be used as a default argument, as it's usage of integrating it with your app is optional.

Reply:
4a19543
> It seems context access can already be used by manager?.activity as elsewhere in the class.

Reply:
What I meant is `ApplicationHolder.applicationContext` can throw an exception if not set (since it's usage is optional in the framework), so it should be replacable by the user.

correct usage:
`SomeClass(context:Context = ApplicationHolder.applicationContext) // can be replaced by the user`
incorrect usage:
`customTabsIntent.launchUrl(ApplicationHolder.applicationContext, url) // impossible for the user to set some other value here`

Reply:
Thanks, now I understand. I pushed the changes where I'm using `manager?.activity`